enhancements/workload-partitioning/wide-availability-workload-partitioning.md

```diff
@@ -0,0 +1,731 @@
+  this feature will be part of the installation phase only.
+
+### Non-Goals
+
+This enhancement expands on the existing [Management Workload
+Partitioning](management-workload-partitioning.md) and as such shares similar
+but slightly different non-goals.
+
+> Note: Items in bold are modifications/additions from the previous enhancement,
+> [Management Workload Partitioning](management-workload-partitioning.md)
+
+- This enhancement is focused on CPU resources. Other compressible resource
+  types may need to be managed in the future, and those are likely to need
+  different approaches.
+- This enhancement does not address non-compressible resource requests, such as
+  for memory.
+- This enhancement does not address ways to disable operators or operands
+  entirely.
+- This enhancement does not address reducing actual utilization, beyond
+  providing a way to have a predictable upper-bounds. There is no expectation
+  that a cluster configured to use a small number of cores for management
+  services would offer exactly the same performance as the default. It must be
+  stable and continue to operate reliably, but may respond more slowly.
+- **This enhancement does not address mixing nodes with pinning and without,
+  this feature will be enabled cluster wide and encapsulate both master and
+  worker pools. If it's not desired then the default behavior will still be
+  turned on but the management workloads will run on the whole CPU set for that
+  desired pool.**
+- **This enhancement assumes that the configuration of a management CPU pool is
+  done as part of installing the cluster. It can be changed after the fact but
+  we will need to stipulate that, that is currently not supported. The intent
+  here is for this to be supported as a day 0 feature, only.**
+- **This enhancement does not address mixing nodes of different CPU sizes in
+  Machine Configuration Pools. Pools must contain machines of the same CPU set size,
+  currently the Performance Profile defines and targets specific machine pools
+  to apply the configuration to, as such they must have the same CPU set size.**
+
+## Proposal
+
+We will need to maintain a global identifier that is set during installation and
+can not be easily removed after the fact. This approach will help remove
+exposing this feature via an API and limit the chances that a misconfiguration
+can cause un-recoverable scenarios for our customers. At install time we will
+also apply an initial machine config for workload partitioning that sets a
+default CPUSet for the whole CPUSet. Effectively this will behave as if workload
+partitioning is not turned on. When a customer wishes to pin the management
+workloads they will be able to do that via the existing Performance Profile.
+Resizing partition size will not cause any issues after installation.
+
+With this approach we eliminate the race condition that can occur if we apply
+the machine config after bootstrap via NTO. Since we create a "default" cri-o
```

Comment:
Sounds good, I'll update with that information

Reply:
What happens when someone tries to use a guaranteed workload on this partially configured cluster? Will this in fact do workload partitioning by moving all infra pods to the all cpus cpuset and removing those from the cpu manager logic? So any workload pod will actually not have dedicated cpus, because infra pods will be there as well?

Reply:
Right in this default state, it behaves as though CPU Partitioning is not enforced because the user hasn't specified which CPUSets they want to limit the workloads to. Side note as well, the current behavior is also not to mutate the pods if they specify a Guaranteed QOS.

Reply:
+1